src/Exception/RequestException.php

```diff
@@ -142,6 +146,24 @@ public static function getResponseBodySummary(ResponseInterface $response)
     }
 
     /**
+     * Obfuscates URI if there is an username and a password present
+     *
+     * @param UriInterface $uri
+     *
+     * @return UriInterface
+     */
+    public static function obfuscateUri($uri)
```

Comment:
Do we need this as public static? I would say make it private and test only the behaviour of the exception without exposing this as a behaviour too.